src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -135,7 +135,7 @@ public bool ChildContentRetrieved
         /// <summary>
         /// <see cref="ITagHelper"/>s that should be run.
         /// </summary>
-        public IEnumerable<ITagHelper> TagHelpers
+        public IList<ITagHelper> TagHelpers
```

Comment:
Oh and I also felt that `Order` was more of a `TagHelperRunner`s responsibility.